examples/hcp-eks-demo/main.tf

```diff
@@ -54,46 +54,54 @@ module "eks" {
 }
 
 # The HVN created in HCP
-resource "hcp_hvn" "main" {
-  hvn_id         = var.hvn_id
-  cloud_provider = "aws"
-  region         = var.hvn_region
-  cidr_block     = var.hvn_cidr_block
+# resource "hcp_hvn" "main" {
+#   hvn_id         = var.hvn_id
+#   cloud_provider = "aws"
+#   region         = var.hvn_region
+#   cidr_block     = var.hvn_cidr_block
+# }
+
+data "hcp_hvn" "example" {
+  hvn_id = "hvn"
 }
 
-module "aws_hcp_consul" {
-  source  = "hashicorp/hcp-consul/aws"
-  version = "~> 0.8.9"
-
-  hvn                = hcp_hvn.main
-  vpc_id             = module.vpc.vpc_id
-  subnet_ids         = module.vpc.private_subnets
-  route_table_ids    = module.vpc.private_route_table_ids
-  security_group_ids = var.install_eks_cluster ? [module.eks[0].cluster_primary_security_group_id] : [""]
-}
-
-resource "hcp_consul_cluster" "main" {
-  cluster_id      = var.cluster_id
-  hvn_id          = hcp_hvn.main.hvn_id
-  public_endpoint = true
-  tier            = var.tier
+# Note: Uncomment the below module to setup peering for connecting to a private HCP Consul cluster
+# module "aws_hcp_consul" {
+#   source  = "hashicorp/hcp-consul/aws"
+#   version = "~> 0.8.9"
+#   hvn                = hcp_hvn.main
+#   vpc_id             = module.vpc.vpc_id
+#   subnet_ids         = module.vpc.private_subnets
+#   route_table_ids    = module.vpc.private_route_table_ids
+#   security_group_ids = var.install_eks_cluster ? [module.eks[0].cluster_primary_security_group_id] : [""]
+# }
```

Comment:
Note: This will remain commented to stay as a reference in case the user wants to setup peering. We could argue that the top level module is clear enough - so open to feedback if we want to remove this.

Reply:
I think it is reasonable to leave for now. 👍